xds/internal/resolver/watch_service_test.go

```diff
@@ -16,375 +16,375 @@
-	xdsC.InvokeWatchRouteConfigCallback(routeStr+"2", xdsresource.RouteConfigUpdate{
-		VirtualHosts: []*xdsresource.VirtualHost{
-			{
-				Domains: []string{targetStr},
-				Routes:  []*xdsresource.Route{{Prefix: newStringP(""), WeightedClusters: map[string]xdsresource.WeightedCluster{cluster + "2": {Weight: 1}}}},
-			},
-		},
-	}, nil)
-	if err := verifyServiceUpdate(ctx, serviceUpdateCh, wantUpdate2); err != nil {
+	t.Cleanup(cleanup)
+
+	// Configure default listener and route configuration resources.
+	resources := e2e.UpdateOptions{
+		NodeID:         nodeID,
+		Listeners:      []*v3listenerpb.Listener{e2e.DefaultClientListener(defaultTestServiceName, defaultTestRouteConfigName)},
+		Routes:         []*v3routepb.RouteConfiguration{e2e.DefaultRouteConfig(defaultTestRouteConfigName, defaultTestServiceName, defaultTestClusterName)},
+		SkipValidation: true,
+	}
+	if err := mgmtServer.Update(ctx, resources); err != nil {
 		t.Fatal(err)
 	}
+
+	// Create a test resolver.ClientConn and a resolver for the test.
+	tcc, _ := buildResolverForTarget(t, resolver.Target{URL: *testutils.MustParseURL("xds:///" + defaultTestServiceName)})
+
+	waitForResourceNames(ctx, t, listenerResourceNamesCh, []string{defaultTestServiceName})
+	waitForResourceNames(ctx, t, routeConfigResourceNamesCh, []string{defaultTestRouteConfigName})
+	verifyUpdateFromResolver(ctx, t, tcc, wantDefaultServiceConfig)
+
+	return mgmtServer, tcc, routeConfigResourceNamesCh
 }
 
-// TestServiceWatchLDSUpdate covers the case that after first LDS and first RDS
-// response, the second LDS response includes a new MaxStreamDuration.  It also
-// verifies this is reported in subsequent RDS updates.
-func (s) TestServiceWatchLDSUpdateMaxStreamDuration(t *testing.T) {
-	serviceUpdateCh := testutils.NewChannel()
-	xdsC := fakeclient.NewClient()
-	cancelWatch := watchService(xdsC, targetStr, func(update serviceUpdate, err error) {
-		serviceUpdateCh.Send(serviceUpdateErr{u: update, err: err})
-	}, nil)
-	defer cancelWatch()
+// Waits for the wantNames to be pushed on to namesCh. Fails the test by calling
+// t.Fatal if the context expires before that.
+func waitForResourceNames(ctx context.Context, t *testing.T, namesCh chan []string, wantNames []string) {
+	t.Helper()
+
+	for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
+		select {
+		case <-ctx.Done():
+		case gotNames := <-namesCh:
+			if cmp.Equal(gotNames, wantNames, cmpopts.EquateEmpty()) {
+				return
+			}
+			t.Logf("Received resource names %v, want %v", gotNames, wantNames)
+		}
+	}
+	t.Fatalf("Timeout waiting for resource to be requested from the management server")
+}
 
+// Tests the case where the listener resource starts pointing to a new route
+// configuration resource after the xDS resolver has successfully resolved the
```

Comment:
Mention old system (i.e. listener resource was already points to a route configuration, and then gets a new update pointing to a new one). And optionally put test name at beginning of comment.